fetch_cluster_logs.bash

```diff
@@ -0,0 +1,170 @@
+#
+# COMMANDS
+#   open          Fetch the logs for an open cluster.
+#   enterprise    Fetch the logs for an enterprise cluster.
+#
+# REQUIRED ARGUMENTS
+#   <ssh-user>            ssh user to access the nodes of your cluster.
+#   <master-public-ip>    Public ip for the master node of your cluster.
+#
+# OPTIONAL ARGUMENTS
+#   --login-token=<token>     Token used to log in to your open DC/OS cluster. If omitted, you will be prompted for it
+#                             during the DC/OS CLI cluster setup.
+#   --username=<username>     Username to log in to your enterprise DC/OS cluster. If omitted, you will be prompted for
+#                             it during the DC/OS CLI cluster setup.
+#   --password=<password>     Password to log in to your DC/OS enterprise cluster. If omitted, you will be prompted for
+#                             it during the DC/OS CLI cluster setup.
+#   --identity-file=<path>    Path to the private ssh key that will be used to ssh into the nodes. If omitted, that key
+#                             must be added to your ssh-agent.
+#   --debug                   Turn on debug logging for the DC/OS CLI.
+
+set +e
+set -x
+
+dcos_variant=$1
+if [[ -z $dcos_variant ]] || ([[ $dcos_variant != "open" ]] && [[ $dcos_variant != "enterprise" ]]); then
+  echo "ERROR: You must specify either the 'open' or 'enterprise' command depending on your cluster type."
+  exit 0
+fi
+shift
+
+ssh_user=$1
+if [[ -z $ssh_user ]]; then
+  echo "ERROR: Required argument 'ssh_user' was either not specified or null."
+  exit 0
+fi
+shift
+
+master_public_ip=$1
+if [[ -z $master_public_ip ]]; then
+  echo "ERROR: Required argument 'master_public_ip' was either not specified or null."
+  exit 0
+fi
+shift
+
+if [[ $dcos_variant == "open" ]]; then
+  for i in "$@"
+  do
+  case $i in
+    --debug)
+      debug_options="--debug --log-level=debug"
+      shift
+    ;;
+    --login-token=*)
+      dcos_login_token_input="<<< ${i#*=}"
+      shift
+    ;;
+      --identity-file=*)
+      identity_file="${i#*=}"
+    shift
+    ;;
+    *)
+      echo "ERROR: Unrecognized argument '$i' for command '$dcos_variant'"
+      exit 0
+    ;;
+  esac
+  done
+else
+  for i in "$@"
+  do
+  case $i in
+    --debug)
+      debug_options="--debug --log-level=debug"
+      shift
+    ;;
+    --username=*)
+      dcos_username=$i
+      shift
+    ;;
+    --password=*)
+      dcos_password=$i
+      shift
+    ;;
+    --identity-file=*)
+      identity_file="${i#*=}"
+      shift
+    ;;
+    *)
+      echo "ERROR: Unrecognized argument '$i' for command '$dcos_variant'"
+      exit 0
+    ;;
+  esac
+  done
+fi
+
+ssh_options="-A -T -l $ssh_user -o StrictHostKeyChecking=no -o UserKnownHostsFile=/dev/null"
+if [[ ! -z $identity_file ]]; then
+  ssh_options="-i $identity_file ${ssh_options}"
+fi
+
+# download the DC/OS CLI
+if [ ! -f dcos-cli ]; then
+  wget https://downloads.dcos.io/binaries/cli/linux/x86-64/dcos-1.12/dcos --output-document=dcos-cli
```

Comment:
The only possible change I can see making is that I could imagine us perhaps needing to change this value up at some point in the future (platform, version, etc).  Perhaps listing this as a variable early (eg DCOS_CLI_VER/URL: <value>) or making this an optional user input that default to this value, but a nice-to-have than necessary.